File: problems/0001-0010/0001-two-sum/hash-lookup/solution.py
"""
Success
Details
Runtime: 48 ms, faster than 79.33% of Python3 online submissions for Two Sum.
Memory Usage: 14.1 MB, less than 62.09% of Python3 online submissions for Two Sum.
"""

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = {}
        for idx, num in enumerate(nums):
            if (target - num) in nums_dict:
                return [
                    nums_dict[target - num],
                    idx,
                ]
            nums_dict[num] = idx

File: problems/0001-0010/0001-two-sum/hash-lookup/solution.js
/**
 * Success
 * Details
 * Runtime: 56 ms, faster than 86.06% of JavaScript online submissions for Two Sum.
 * Memory Usage: 36.2 MB, less than 11.99% of JavaScript online submissions for Two Sum.
 */

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let lookup_table = new Object();
    let result = [];
    nums.forEach((num, idx) => {
        if ((target - num) in lookup_table) {
            result = [
                lookup_table[target - num],
                idx,
            ];
        }
        lookup_table[num] = idx;
    });
    return result;
};
